src/sentry/shared_integrations/client/base.py

```diff
@@ -143,19 +144,20 @@ def _request(
                 self.track_response_data("timeout", span, e)
                 raise ApiTimeoutError.from_exception(e) from e
             except HTTPError as e:
-                resp = e.response
-                if resp is None:
-                    self.track_response_data("unknown", span, e)
-
-                    # It shouldn't be possible for integration_type to be null.
-                    extra = {"url": full_url}
-                    if self.integration_type:
-                        extra[self.integration_type] = self.name
-                    self.logger.exception("request.error", extra=extra)
-
-                    raise ApiError("Internal Error", url=full_url) from e
-                self.track_response_data(resp.status_code, span, e)
-                raise ApiError.from_response(resp, url=full_url) from e
+                self._handle_HTTPError(e, full_url, span)
+            except Exception as e:
+                # Sometimes a ConnectionResetError shows up two or three deep in an exception
+                # chain, and you end up with an exception like
+                #     `ChunkedEncodingError("Connection broken: ConnectionResetError(104, 'Connection reset by peer')",
+                #          ConnectionResetError(104, 'Connection reset by peer'))`,
+                # which is a ChunkedEncodingError caused by a ProtocolError caused by a ConnectionResetError.
+                # Rather than worrying about what the other layers might be, we just stringify to detect this.
+                if "ConnectionResetError" in str(e):
```

Comment:
Nice trick 👍🏻 

Would it make sense to do handle `ConnectionResetError` within `_handle_HTTPError` and have the code look like this?

```
if "ConnectionResetError" in str(e) or "InvalidChunkLength" in str(e):
    self._handle_HTTPError(e, full_url, span)
```

Whatever you prefer.